pkg/pma/page.h

```diff
@@ -0,0 +1,13 @@
+/// @file
+
+#ifndef PMA_PAGE_H
+#define PMA_PAGE_H
+
+#include <stddef.h>
+
+#include "util.h"
+
+/// Number of bytes in a page.
+#define kPageSz ((size_t)KiB(16))
```

Comment:
The way I've been thinking about this is that our page size should be at least as large as the page size of the platform with the largest page size we want to support to keep things conceptually simple. For example, if our page size was 4KB, and we're running on an Apple Silicon Mac, which has 16KB pages: we pass in 4KB to `mmap()` or `mprotect()`, both of which silently extend their effects to the entire surrounding 16KB page, which is fine but non-obvious from looking at the code. On the other hand, when our page size is at least as large the system page size, `mmap()` and `mprotect()` will operate on the length we're passing them rather than silently rounding them up. Arguably, we should be using a 64KB page. Larger pages also mean fewer SIGSEGV we have to handle.

Reply:
Thanks for explaining the reasoning. I didn't realize it didn't really have anything to do with the loom having 4 byte granularity.

It's true that larger pages mean fewer sigsegv (though more data to write on snapshot). I'm curious what might be the optimum here. Not that it really matters for this PR.

Reply:
IIRC there's an architecture that uses 16KB pages (M1 macs?), so we just do 16KB everywhere.